ksql-engine/src/main/java/io/confluent/ksql/structured/SchemaKStream.java

```diff
@@ -286,8 +287,43 @@ private boolean rekeyRequired(List<Expression> groupByExpressions) {
         && fieldNameFromExpression(groupByExpressions.get(0)).equals(keyFieldName));
   }
 
+  static Pair<String, List<Integer>> keyIndexesForGroupBy(
+      final Schema schema, final List<Expression> groupByExpressions) {
+    final StringBuilder aggregateKeyName = new StringBuilder();
+    final List<Integer> newKeyIndexes = new ArrayList<>();
+
+    // Collect the column indexes, and build the new key as <column1>+<column2>+...
+    boolean addSeparator = false;
+    for (Expression groupByExpr : groupByExpressions) {
```

Comment:
I think using streams here would be cleaner:

```java
final String aggregateKeyName = groupByExpressions.stream()
        .map(Expression::toString)
        .collect(Collectors.joining(GROUP_BY_COLUMN_SEPARATOR));
    
groupByExpressions.forEach(e -> SchemaUtil.getIndexInSchema(e.toString(), schema));
```

(I realize you've only refactored existing code into a method, so feel free to ignore if you want as you've already improved the code :D

Reply:
Makes sense to me. I'll write it with a stream.